daemon/initlayer/setup_unix.go

```diff
@@ -58,8 +58,12 @@ func Setup(initLayerFs containerfs.ContainerFS, rootIDs idtools.IDPair) error {
 					f.Chown(rootIDs.UID, rootIDs.GID)
 					f.Close()
 				default:
-					if err := os.Symlink(typ, filepath.Join(initLayer, pth)); err != nil {
-						return err
+					linkpath := filepath.Join(initLayer, pth)
+					// Replace pth only if it is not a symbolic link
+					if _, err := os.Readlink(linkpath); err != nil {
```

Comment:
1. We already did `Stat()` on this above in line 43, can use it here instead of calling `Readlink()`.
2. There are too many calls to `filepath.Join(initLayer, pth)` here, perhaps it can be consolidated?

In other words, something like this (untested):
```go
 		dst := filepath.Join(initLayer, pth)
 		if st, err := os.Stat(dst); err != nil {
...
 		 		default:
 		 		 		// Replace pth only if it is not a symbolic link
 		 		 		if st.Mode()&os.ModeSymlink != 0 {
 		 		 		 		if err := os.Symlink(typ, dst); err != nil {
 		 		 		 		 		return err
 		 		 		 		}
 		 		 		}
...
```